src/MLJInterface.jl

```diff
@@ -0,0 +1,231 @@
+            return $constructor
+        end
+    end
+end
+function getsymb(ex::Symbol)
+    return ex
+end
+function getsymb(ex::Expr)
+    for arg in ex.args
+        isa(arg, Symbol) && return arg
+        s = getsymb(arg)
+        isa(s, Symbol) && return s
+    end
+    return nothing
+end
+
+"""Get an equivalent `Options()` object for a particular regressor."""
+function get_options(::AbstractSRRegressor) end
+
+eval(modelexpr(:SRRegressor))
+eval(modelexpr(:MultitargetSRRegressor))
+
+# Cleaning already taken care of by `Options` and `equation_search`
+function full_report(m::AbstractSRRegressor, fitresult)
+    _, hof = fitresult.state
+    # TODO: Adjust baseline loss
+    formatted = format_hall_of_fame(hof, fitresult.options, 1.0)
+    equation_strings = get_equation_strings_for(m, formatted.trees, fitresult.options)
+    best_idx = dispatch_selection_for(
+        m, formatted.trees, formatted.losses, formatted.scores, formatted.complexities
+    )
+    return (;
+        best_idx=best_idx,
+        equations=formatted.trees,
+        equation_strings=equation_strings,
+        losses=formatted.losses,
+        complexities=formatted.complexities,
+        scores=formatted.scores,
+    )
+end
+
+# TODO: Pass `variable_names` and `units`
+# TODO: Enable `verbosity` being passed to `equation_search`
+MMI.clean!(::AbstractSRRegressor) = ""
+
+function MMI.fit(m::AbstractSRRegressor, verbosity, X, y, w=nothing)
+    return _update(m, verbosity, (; state=nothing), nothing, X, y, w)
+end
+function MMI.update(
+    m::AbstractSRRegressor, verbosity, old_fitresult, old_cache, X, y, w=nothing
+)
+    return _update(m, verbosity, old_fitresult, old_cache, X, y, w)
+end
+function _update(m::AbstractSRRegressor, verbosity, old_fitresult, old_cache, X, y, w)
+    options = get(old_fitresult, :options, get_options(m))
+    mX = MMI.matrix(X)
+    variable_names = getcolnames(X)
+    X_t = copy(transpose(mX))
+    y_t = format_input_for(m, y)
+    search_state = equation_search(
+        X_t,
+        y_t;
+        niterations=m.niterations,
+        weights=w,
+        variable_names=variable_names,
+        options=options,
+        parallelism=m.parallelism,
+        numprocs=m.numprocs,
+        procs=m.procs,
+        addprocs_function=m.addprocs_function,
+        runtests=m.runtests,
+        saved_state=old_fitresult.state,
+        return_state=true,
+        loss_type=m.loss_type,
+    )
+    fitresult = (; state=search_state, options=options)
+    return (fitresult, nothing, full_report(m, fitresult))
+end
+
+function getcolnames(X)
+    try
+        return getcolnames(MMI.schema(X), X)
+    catch e
+        isa(e, ArgumentError) || rethrow(e)
+    end
+    try
+        return getcolnames(MMI.schema(MMI.table(X)), X)
+    catch e
+        isa(e, MethodError) || isa(e, ArgumentError) || rethrow(e)
+    end
+    return getcolnames(nothing, X)
+end
+getcolnames(::Nothing, X) = [map(i -> "x$(i)", axes(X, 2))...]
```

Comment:
Also, the thing is that not all tables have a schema defined on the fly.

Reply:
I see. So what do you think the best solution is? Will this work, because of the `try-catch`'s used above?

Reply:
``` 
function get_matrix_and_colnames(X)
    sch = MMI.istable(X) ? MMI.schema(X) : nothing
    Xm_t = MMI.matrix(X, transpose=true)
    colnames = sch === nothing ? [map(i -> "x$(i)", axes(Xm_t, 1))...] : [string.(sch.names)...]
    return  Xm_t, colnames
end
```

Reply:
Nice, looks good!

Reply:
Implemented.